monai/auto3dseg/utils.py

```
@@ -368,13 +407,109 @@ def algo_from_pickle(pkl_filename: str, template_path: PathLike | None = None, *
+
+    # Try each template path to instantiate the class
+    paths_to_try: list[str | None] = template_paths if template_paths else [None]
+    algo = None
+    used_template_path: str | None = None
+    for path in paths_to_try:
+        try:
+            if path and path not in sys.path:
+                sys.path.insert(0, path)
+
+            algo_config: dict[str, Any] = {"_target_": target}
+            if "template_path" in state and state["template_path"]:
```

Ruff (0.14.10)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)